pkg/ottl/ottlfuncs/README.md

```diff
@@ -58,6 +59,21 @@ Available Editors:
 - [set](#set)
 - [truncate_all](#truncate_all)
 
+### append
+
+`append(target, Optional[value], Optional[values])`
+
+The `append` function appends single or multiple string values to `target`. 
+`append` converts scalar values into an array if the field exists but is not an array, and creates an array containing the provided values if the field doesn’t exist.
+
+Resulting field is always of type `pcommon.Slice` and will not convert the types of existing or new items in the slice. This means that it is possible to create a slice whose elements have different types.  Be careful when using `append` to set attribute values, as OTLP does not support slices as attribute values when the elements in the slice are not all the same type.
+
+  - `append(attributes["tags"], "prod")`
+  - `append(attributes["tags"], values = ["staging", "staging:east"])`
+  - `append(attributes["tags_copy"], attributes["tags"])`
+
+`append` is able to create heterogenous slices that are not supported by Open Telemetry attributes.
```

Comment:
```suggestion
```

We have this addressed above now, so we should try to only mention it once. I do think it is worth mentioning earlier.